tests/i18n/tests.py

```diff
@@ -1859,6 +1859,16 @@ def test_page_with_dash(self):
         response = self.client.get('/de-simple-page/')
         self.assertEqual(response.content, b'Yes')
 
+    def test_no_redirect_on_404(self):
+        """
+        Regression test for #27402.
```

Comment:
Could you please update the docstring as per the "In test docstrings, ..." instructions at https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style (which were added yesterday, so please let me know if anything is unclear).